lib/ZwaveClient.js

```diff
@@ -17,7 +17,7 @@ const { socketEvents } = reqlib('/lib/SocketManager.js')
 const store = reqlib('config/store.js')
 const storeDir = utils.joinPath(true, reqlib('config/app.js').storeDir)
 const debug = reqlib('/lib/debug')('Zwave')
-const logger = require('./logger.js').module('ZWC')
+const logger = require('./logger.js').setup('zwc')
```

Comment:
I would prefer to use `module` to get the module and a separete function `setup` to edit the general configuration.

Reply:
Do you mean something like this?
```javascript
const logger = require('./logger.js').module('Zwave').setup(
  { enabled: true, level: 'info', logToFile: true, filename: "zwavejs2mqtt.log" }
)
```

Reply:
@ahochsteger Yep and we should decide where and when we will call the setup. It should be called only on application start and when settings change. By default module will return the logger with the last setup configuration.

Reply:
@robertsLando it's done now and can be used like this to dynamically reconfigure the logger:

```javascript
let logger = require('./logger.js').module('Zwave')
log.info('This will be shown by default')
logger = logger.setup({ level: 'warn' })
log.info('Now this won't be shown anymore')
```

Since Winston does not seem to support the reconfiguration of an existing logger during runtime I had to do some workarounds which I'm not really happy about yet:
`setup()` closes an existing logger with the given module name and creates a completely new one, thus all references will change and `const logger = ...` will have to be replaced with `let logger = ...`.

Maybe somebody with more Winston knowledge is able to find a better way.

Reply:
Update: Have a look at the section *Open questions* in the description for possible solutions.

Reply:
Another update: Winston supports reconfiguration of existing loggers using `logger.configure()` which I was not aware before. That way it was really simple to do it the way we need it.